### PR TITLE
Don't output links in the Top Earning Pages widget when on View-only Dashboard.

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue:

- #8451

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.js [hunk 1]
```diff
@@ -295,7 +300,11 @@ function DashboardTopEarningPagesWidgetGA4( {
 	];
 
 	return (
-		<Widget noPadding Footer={ Footer } ref={ updateTrackingRef }>
+		<Widget
+			noPadding
+			Footer={ ! viewOnlyDashboard ? Footer : undefined }
```
Comment (kuasha420) [inline assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.js:305]: This is actually unnecessary, as the `SourceLink` used in `Footer` will return `null` on view-only dashboard. It results in an empty footer which is consistent look-wise with other widgets. See `DashboardPopularKeywordsWidget` for example.

**Review by kuasha420 — CHANGES_REQUESTED** [review]
@tofumatt Thanks. Looks good to me for the most part, just one thing I left a note about. Please have a look at let me know what you think. 

Cheers.

File: assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.js (hunk 1 above)
Comment (tofumatt) [reply assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.js:305]: Ah, indeed it does… I noticed that, but then saw there was a needless empty footer section when using the `Footer` component in view-only mode...

But, that's consistent with other widgets: https://google.github.io/site-kit-wp/storybook/develop/?path=/story/modules-searchconsole-widgets-searchfunnelwidgetga4--view-only-search-console-only-ready

So I guess it's a bigger change and I won't do it here.

Comment (tofumatt) [reply assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.js:305]: Weirdly now I can't even reproduce that! 😓

Alright, I've made the change 👍🏻

**Pull request merged**